OutOfSchool/OutOfSchool.WebApi.Tests/Validators/CustomAgeAttributeTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using NUnit.Framework;
+using OutOfSchool.Common.Validators;
+
+namespace OutOfSchool.WebApi.Tests.Validators;
+
+[TestFixture]
+public class CustomAgeAttributeTests
+{
+    [Test]
+    public void IsValid_WhenDateIsNull_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = null as DateTime?;
+        var customDayOfBirthAttribute = new CustomAgeAttribute();
+
+        // Act
+        var isValid = customDayOfBirthAttribute.IsValid(dateOfBirth);
```

Comment:
(Test design) You could leave the code as it is now. I'm just suggesting improvements, but don't insist.

More code doesn't mean better code. "Arrange-Act-Assert" test pattern doesn't mean you should always have each separate part in your test. Sometimes simpler code is also easy to read. 

For this specific test you could omit "Arrange" part like 
```
        // Act
        var isValid = new CustomAgeAttribute().IsValid(null as DateTime?);
```
3 lines of code less, same meaning.

I agree, that sometimes system-under-test creation is more complex and could be done in "Arrange" part (i.e. in later tests you have `new CustomAgeAttribute() { MinAge = -1 }`), but not this time. Same for input data.

Reply:
> ... collapsing similar tests ...

It's ok, but only if they are similar - describing one test case. (In other words, if test fails for every case it should be due to problems in same piece of implementation code). Re your specific example - test named `...WhenInputDataValid...` has too many acceptance criteria and thus I would disagree with naming and implementation if it relates to naming.

I.e. - check `minAge =0, maxAge=17` for `17 years`, `17 years and 2 days`, `18 years minus 1 day` -  seems similar, but `0 years`, `5 years`, `null years` - seems less similar.

Btw, fake test data generators usually provides better way to get more test inputs for specific test case. It is not achieved in one test run, it requires better test design (more code to configure faker), but in long run it is better.

Reply:
> > ... collapsing similar tests ...
> 
> It's ok, but only if they are similar - describing one test case. (In other words, if test fails for every case it should be due to problems in same piece of implementation code). Re your specific example - test named `...WhenInputDataValid...` has too many acceptance criteria and thus I would disagree with naming and implementation if it relates to naming.
> 
> I.e. - check `minAge =0, maxAge=17` for `17 years`, `17 years and 2 days`, `18 years minus 1 day` - seems similar, but `0 years`, `5 years`, `null years` - seems less similar.
> 
> Btw, fake test data generators usually provides better way to get more test inputs for specific test case. It is not achieved in one test run, it requires better test design (more code to configure faker), but in long run it is better.

Thanks! I have already fixed some test design, but this changes with collapsing similar tests I will leave to another PR, because as you said it requires better test design and it can lead to change whole bunch of tests that already implemented.